.github/workflows/initialize-app.yml

```diff
@@ -1,71 +1,82 @@
           - staging
+          - production
 
       edge_app_name:
-        description: 'Name of the edge app'
+        description: 'Folder name of the edge app (under edge-apps/) to initialize'
         required: true
         default: ''
-        type: string
 
       edge_app_title:
-        description: 'Title of the edge app'
+        description: 'Display title of the edge app (used in instance.yml)'
         required: true
         default: ''
-        type: string
+
+run-name: Initializing ${{ inputs.edge_app_name }} in ${{ inputs.environment }}
 
 jobs:
   deploy:
-    environment: ${{ github.event.inputs.environment }}
+    name: Initialize App in ${{ inputs.environment }}
     runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
     env:
-      APP_PATH: 'edge-apps/fake-dashboard'
+      APP_PATH: edge-apps/${{ inputs.edge_app_name }}
+      MANIFEST_FILE_NAME: ${{ inputs.environment == 'production' && 'screenly.yml' || 'screenly_qc.yml' }}
 
     steps:
-      - name: Checkout Code
+      - name: Checkout master branch
         uses: actions/checkout@v4
+        with:
+          ref: master
 
       - name: Create Edge App
         uses: screenly/cli@master
         with:
           screenly_api_token: ${{ secrets.SCREENLY_API_TOKEN }}
-          # yamllint disable-line rule:line-length
-          cli_commands: edge-app create --in-place --path=edge-apps/${{ inputs.edge_app_name }}
+          cli_commands: edge-app create --in-place --path=${{ env.APP_PATH }}
 
       - name: Deploy Edge App
         uses: screenly/cli@master
         with:
           screenly_api_token: ${{ secrets.SCREENLY_API_TOKEN }}
-          # yamllint disable-line rule:line-length
-          cli_commands: edge-app deploy --path=edge-apps/${{ inputs.edge_app_name }}
+          cli_commands: edge-app deploy --path=${{ env.APP_PATH }}
 
       - name: Create Edge App Instance
         uses: screenly/cli@master
         with:
           screenly_api_token: ${{ secrets.SCREENLY_API_TOKEN }}
-          # yamllint disable-line rule:line-length
-          cli_commands: edge-app instance create --path=edge-apps/${{ inputs.edge_app_name }}
+          cli_commands: edge-app instance create --path=${{ env.APP_PATH }}
 
       - name: Correct the instance name
         run: |
-          cd edge-apps/${{ inputs.edge_app_name }}
-          sed -i "s/^name: .*/name: ${{ inputs.edge_app_title }}/" instance.yml
+          sed -i "s/^name: .*/name: ${{ inputs.edge_app_title }}/" ${{ env.APP_PATH }}/instance.yml
 
       - name: Update Edge App Instance
         uses: screenly/cli@master
         with:
           screenly_api_token: ${{ secrets.SCREENLY_API_TOKEN }}
-          cli_commands: edge-app instance update --path=edge-apps/${{ inputs.edge_app_name }}
+          cli_commands: edge-app instance update --path=${{ env.APP_PATH }}
+
+      - name: Install yq (YAML parser)
+        run: |
+          sudo apt-get update && sudo apt-get install -y yq
```

Comment:
The Debian package for `yq` is [no longer maintained](https://github.com/mikefarah/yq?tab=readme-ov-file#community-supported-installation-methods), but you can keep this for now as long as it works.

You could [download the latest binary](https://github.com/mikefarah/yq/releases/latest) (**yq_linux_amd64**) instead.